prime-router/metadata/fhir_mapping/hl7/secondary/Destination.yml

```diff
@@ -1,70 +1,135 @@
 name:
+  condition: $msh25 NOT_NULL
   type: STRING
-  valueOf: MSH.5
+  valueOf: MSH.5.1
   expressionType: HL7Spec
+  vars:
+    msh25: MSH.25
 
-endpoint:
+endpoint_1:
+  condition: $msh25 NULL && $idStr NOT_NULL && $systemStr EQUALS ISO
   type: URL
-  required: true
-  valueOf: MSH.25 | MSH.6
+  valueOf: $idValue
   expressionType: HL7Spec
+  vars:
+    msh25: MSH.25
+    idStr: MSH.5.2
+    idValue: $urlPrefix + MSH.5.2
+    systemStr: String, MSH.5.3
+  constants:
+    urlPrefix: "urn:oid:"
 
-# For OIDs
-extension_1:
-  generateList: true
-  valueOf: extension/Extension
-  expressionType: resource
-  condition: $idStr NOT_NULL && $systemStr EQUALS ISO
+endpoint_2:
+  condition: $msh25 NULL && $idStr NOT_NULL && $systemStr EQUALS UUID
+  type: URL
+  valueOf: $idValue
+  expressionType: HL7Spec
   vars:
-    idStr: String, MSH.5.2
-    value: $oidUrlPrefix + MSH.5.2
+    msh25: MSH.25
+    idStr: MSH.5.2
+    idValue: $urlPrefix + MSH.5.2
     systemStr: String, MSH.5.3
   constants:
-    oidUrlPrefix: "urn:oid:"
-    KEY_NAME_SUFFIX: Oid
-    urlValue: application-identifier
+    urlPrefix: "urn:uuid:"
 
-# For all other ID types
-extension_2:
-  generateList: true
-  valueOf: extension/Extension
-  expressionType: resource
-  condition: $idStr NOT_NULL && $systemStr NOT_EQUALS ISO
+endpoint_3:
+  condition: $msh25 NULL && $idStr NOT_NULL && $systemStr EQUALS DNS
+  type: URL
+  valueOf: $idValue
+  expressionType: HL7Spec
   vars:
-    idStr: String, MSH.5.2
-    value: $idUrlPrefix + MSH.5.2
+    msh25: MSH.25
+    idStr: MSH.5.2
+    idValue: $urlPrefix + MSH.5.2
     systemStr: String, MSH.5.3
   constants:
-    idUrlPrefix: "urn:id:"
-    KEY_NAME_SUFFIX: String
-    urlValue: application-identifier
+    urlPrefix: "urn:dns:"
 
-# For OIDs
-extension_3:
-  generateList: true
-  valueOf: extension/Extension
-  expressionType: resource
-  condition: $idStr NOT_NULL && $systemStr EQUALS ISO
+endpoint_4:
+  condition: $msh25 NULL && $idStr NOT_NULL && $systemStr EQUALS URI
+  type: URL
+  valueOf: $idValue
+  expressionType: HL7Spec
+  vars:
+    msh25: MSH.25
+    idStr: MSH.5.2
+    idValue: $urlPrefix + MSH.5.2
+    systemStr: String, MSH.5.3
+  constants:
+    urlPrefix: "urn:uri:"
+
+endpoint_5:
+  condition: $msh25 NULL && $idStr NOT_NULL && $systemStr EQUALS CLIA
+  type: URL
+  valueOf: $idValue
+  expressionType: HL7Spec
   vars:
+    msh25: MSH.25
+    idStr: MSH.5.2
+    idValue: $urlPrefix + MSH.5.2
+    systemStr: String, MSH.5.3
+  constants:
+    urlPrefix: "urn:clia:"
+
+endpoint_6:
+  condition: $msh25 NULL && $idStr NOT_NULL && $systemStr NOT_EQUALS UUID && $systemStr NOT_EQUALS ISO && $systemStr NOT_EQUALS DNS && $systemStr NOT_EQUALS URI && $systemStr NOT_EQUALS CLIA
+  type: URL
```

Comment:
Do these conditions map to the conditions in the mapping document? How did you figure out you needed to handle the different types of endpoints?

Reply:
Not directly. This is more like coding to generate the proper data.  These conditions relate to the use of the different ID types like DNS, URI, and how we need to store the data.